packages/editor/src/components/collab-sidebar/hooks.js

```diff
@@ -111,3 +119,128 @@ export function useBlockComments( postId ) {
 
 	return { resultComments, unresolvedSortedThreads, totalPages };
 }
+
+export function useBlockCommentsActions() {
+	const { createNotice } = useDispatch( noticesStore );
+	const { saveEntityRecord, deleteEntityRecord } = useDispatch( coreStore );
+	const { getCurrentPostId } = useSelect( editorStore );
+	const { getBlockAttributes, getSelectedBlockClientId } =
+		useSelect( blockEditorStore );
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const onError = ( error ) => {
+		const errorMessage =
+			error.message && error.code !== 'unknown_error'
+				? decodeEntities( error.message )
+				: __( 'An error occurred while performing an update.' );
+		createNotice( 'error', errorMessage, {
+			type: 'snackbar',
+			isDismissible: true,
+		} );
+	};
+
+	const onCreate = async ( { content, parent } ) => {
+		try {
+			const savedRecord = await saveEntityRecord(
+				'root',
+				'comment',
+				{
+					post: getCurrentPostId(),
+					content,
+					comment_type: 'block_comment',
+					comment_approved: 0,
+					parent: parent || 0,
+				},
+				{ throwOnError: true }
+			);
+
+			// If it's a main comment, update the block attributes with the comment id.
+			if ( ! parent && savedRecord?.id ) {
+				const clientId = getSelectedBlockClientId();
+				const metadata = getBlockAttributes( clientId )?.metadata;
+				updateBlockAttributes( clientId, {
+					metadata: {
+						...metadata,
+						commentId: savedRecord.id,
+					},
+				} );
+			}
+
+			createNotice(
+				'snackbar',
+				parent
+					? __( 'Reply added successfully.' )
+					: __( 'Comment added successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+
+			return savedRecord;
+		} catch ( error ) {
+			onError( error );
+		}
+	};
+
+	const onEdit = async ( { id, content, status } ) => {
+		const messageType = status ? status : 'updated';
+		const messages = {
+			approved: __( 'Comment marked as resolved.' ),
+			hold: __( 'Comment reopened.' ),
+			updated: __( 'Comment updated.' ),
+		};
+
+		try {
+			await saveEntityRecord(
+				'root',
+				'comment',
+				{
+					id,
+					content,
+					status,
+				},
+				{ throwOnError: true }
+			);
+			createNotice(
+				'snackbar',
+				messages[ messageType ] ?? __( 'Comment updated.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+		} catch ( error ) {
+			onError( error );
+		}
+	};
+
+	const onDelete = async ( comment ) => {
+		try {
+			await deleteEntityRecord(
+				'root',
+				'comment',
+				comment.id,
+				undefined,
+				{
+					throwOnError: true,
+				}
+			);
+
+			if ( ! comment.parent ) {
+				updateBlockAttributes( getSelectedBlockClientId(), {
+					blockCommentId: undefined,
+				} );
```

Comment:
```suggestion
				const clientId = getSelectedBlockClientId();
				const metadata = getBlockAttributes( clientId )?.metadata;
				updateBlockAttributes( clientId, {
					metadata: {
						...metadata,
						commentId: undefined,
					},
				} );
```

I think the `blockCommentId` attribute is used nowhere now.

Reply:
Good catch. I created an original branch before metadata migration and missed this when resolving conflicts.